### DISCUSSION TITLE
Will `lampepfl/dotty` move to `scala/scala3`?

### DISCUSSION THREAD

**Discussion by ckipp01** [scala/scala3#17965] [General Discussion]
There have been a handful of times on the Scala Discord and elsewhere I've see questions about where the actual Scala 3 codebase is. For anyone unfamiliar with the history of Dotty the obvious place for this would be under the [Scala Org](https://github.com/scala), so they head there only to be confused. After it's pointed out to them that it lives in `lampepfl/dotty` the next question is almost always **why**? I've seen this question pretty often and never gotten a convincing answer as to why this migration never happened or isn't already in progress. I may be remembering incorrectly, but I thought this _was_ the plan in the past. @SethTisue's recent comment in [here](https://github.com/scala/scala-dev/issues/661#issuecomment-1588397652) made me think of this again and create this discussion. I'd like a singular place where I can link to in the future to either explain why this isn't happening or to show that it's being discussed. So basically:

1. Is the plan to migrate this repo to `scala/scala3` or something under the [Scala Org](https://github.com/scala)
2. If yes, then is there a timeline in mind, or what would be needed to make this happen?
3. If no, why?

**Comment by bjornregnell** [comment]
I completely agree! Every time I've been away from issues for some weeks I forget were things are scattered and have difficulties in remembering where to look. It would be marvelous if all threads in history were discoverable from the [github.com/scala](https://github.com/scala/) org.

**Comment by SethTisue** [comment]
as I see it, the big two reasons to make the change are:

* reduce user confusion (making it likelier that people will find existing issues and report new ones)
* allow unified search across all scala/* repos (GitHub supports organization-level search and it's incREDibly useful)

**Reply by SethTisue** [reply]
A third reason is to improve public perception of Scala 3. Having the repo in the expected location would signal to users and decision makers that the project is mature and well-managed.

**Comment by nicolasstucki** [comment]
One of the reasons we did not do that in the past was that we did not have an exhaustive list of tools that would need to be updated. We could start compiling one here.

**Reply by SethTisue** [reply]
I wonder if much actually needs to be updated, because GitHub will typically redirect. Of course we should `git grep lampepfl/dotty` in the website repos and update what we find, but the links wouldn't 404 even if we didn't. Regardless, I would be happy to take care of the websites, and I can also trawl around scala/* (including scala/scala) looking for stuff to update.

I expect almost anything needing updating would be stuff that mainly compiler hackers use, and if something breaks for us it will probably be obvious what the source of the breakage is.

Can you think of anything else needing updating...? I'm not only eager to see this happen, but eager to help.

**Comment by He-Pin** [comment]
Or we can create a project `scala3` and point it to dotty for now in the Readme and description.

**Comment by He-Pin** [comment]
As scala 3.3.1 is released, I think it would fine to do it now, and I saw people starting to drop scala 2.12 too.

**Comment by SethTisue** [comment]
This was discussed at the Scala core meeting yesterday and it was agreed that lampepfl/dotty will be moved to scala/scala3 after the 3.4.0 release.

**Comment by som-snytt** [comment]
Apparently, it was a smooth transition thanks to the team's attention to detail.

https://github.com/scala/scala3

At least, I have no traumatic recollection of an infamous date on which everything stopped working.
